localstack/aws/handlers/service_plugin.py

```diff
@@ -43,8 +44,8 @@ def require_service(self, _: HandlerChain, context: RequestContext, response: Re
         if service_operation in request_router.handlers:
             return
 
-        # FIXME: this blocks all requests to other services, so a mutex list per-service would be useful
-        with self.mutex:
+        self.service_locks.setdefault(context.service.service_name, threading.RLock())
```

Comment:
since python does not have lazy evaluation, we're creating a new `threading.RLock()` on every call. maybe a factory-style datastructure like the defaultdict would be more appropriate?